crates/matrix-sdk/src/client/builder.rs

```diff
@@ -575,8 +578,13 @@ async fn discover_homeserver_from_server_name_or_url(
     // trying a homeserver URL.
     if let Ok(homeserver_url) = Url::parse(&server_name_or_url) {
         // Make sure the URL is definitely for a homeserver.
-        if check_is_homeserver(&homeserver_url, http_client).await {
-            return Ok((homeserver_url.to_string(), None));
+        match get_supported_versions(&homeserver_url, http_client).await {
+            Ok(_) => {
+                return Ok((homeserver_url, None, None));
```

Comment:
The whole point is to be able to forward the versions response here, right? so `Ok(response) => return Ok((homeserver_url, None, response))`?